python/whylogs/extras/image_metric.py

```diff
@@ -0,0 +1,152 @@
+    """
+    Compute statistics data for a PIL Image
+
+    Args:
+        img (ImageType): PIL Image
+
+    Returns:
+        Dict: of metadata
+    """
+
+    stats = Stat(img.convert("HSV"))
+    metadata = {}
+    for index in range(len(channels)):
+        for statistic_name in image_stats:
+            if hasattr(stats, statistic_name):
+                metadata[channels[index] + "." + statistic_name] = getattr(stats, statistic_name)[index]
+
+    return metadata
+
+
+def get_pil_image_metadata(img: ImageType) -> Dict:
+    """
+    Grab statistics data from a PIL ImageStats.Stat
+
+    Args:
+        img (ImageType): PIL Image
+
+    Returns:
+        Dict: of metadata
+    """
+    metadata = {}
+    for k, v in dict(img.getexif()).items():
+        try:
+            if isinstance(v, IFDRational):
+                metadata[TAGS[k]] = "{}".format(v)
+            else:
+                metadata[TAGS[k]] = v
+        except KeyError:
+            logger.warning(f"Couldn't read exif tag: {k} skipping.")
+
+    metadata.update(image_based_metadata(img))
+    metadata.update(get_pil_image_statistics(img))
+
+    return metadata
+
+
+def image_based_metadata(img):
+    return {
+        "ImagePixelWidth": img.width,
+        "ImagePixelHeight": img.height,
+        "Colorspace": img.mode,
+    }
+
+
+
+@dataclass
+class ImageMetric(CompoundMetric):
+
+    def __init__(self, submetrics: Dict[str, Metric]):
+        super(ImageMetric, self).__init__(submetrics)
+
+    @property
+    def namespace(self) -> str:
+        return "image"
+
+    def columnar_update(self, view: PreprocessedColumn) -> OperationResult:
+        submetric_names = self.submetrics.keys()
+        for image in list(chain.from_iterable(view.raw_iterator())):
+            if isinstance(image, ImageType):
+                metadata = get_pil_image_metadata(image)
+                for attr in metadata.keys():
+                    if attr in submetric_names:
+                        self.submetrics[attr].columnar_update(PreprocessedColumn.apply([metadata[attr]]))
+        return OperationResult.ok(1)
+
+    @classmethod
+    def zero(cls, config: MetricConfig) -> "ImageMetric":
+        return ImageMetric(submetrics={
+            "ImageWidth": DistributionMetric.zero(config),
+            "ImageLength": DistributionMetric.zero(config),
+            "Hue.mean": DistributionMetric.zero(config),
+            "Saturation.mean": DistributionMetric.zero(config),
+            "Brightness.mean": DistributionMetric.zero(config),
+        })
+
+
+def log_image(
+    profiles: Union[List[DatasetProfile],DatasetProfile],
+    images: Union[Dict[str, ImageType], ImageType],
+    default_column_name: str="image",
+) -> None:
+    profiles = profiles if isinstance(profiles, list) else [profiles]
+    images = images if isinstance(images, dict) else {default_column_name: images}
+    for profile in profiles:
+        profile.track(row=images)
```

Comment:
I'm not sure how this plugs into the integration. I think we could support a method like this, but we should return a ResultSet and not require the profiles be passed in, and it would wrap something like your why.log(....) with custom schema kind of thing.